src/conduit/conftest.py

```diff
@@ -104,16 +105,9 @@ def democontent(app_env: AppEnvType, db: Session) -> t.Generator:
     try:
         yield db
     finally:
-
         with transaction.manager:
-            engine = app_env["registry"].settings["sqlalchemy.engine"]
-            engine.execute("TRUNCATE articles, comments, tags, users CASCADE")
-
-            # TODO: Every time someone adds a table they have to remember to
-            # include it here otherwise they might get errors.UniqueViolation
-            # errors in their test suite. It's probably possible to run an
-            # SQL command that would first fetch all table names, then run
-            # TRUNCATE on them.
+            Base.metadata.drop_all()
+            Base.metadata.create_all()
```

Comment:
`create_all` shouldn't be needed as tables are created by our alembic migration steps?

Reply:
The alembic migration fixture, `app_env`, is scoped to session. This would be true if we reduced it to test scope, but otherwise we'd need to recreate. It's a good point, though. This could potentially mask issues where the alembic migration drifts out of sync with the models.

Reply:
Are you sure this is the case? How does it work currently then? We have the SQL to drop tables, something needs to recreate them?

Reply:
Ah of course, the SQL removes the rows, but keeps the tables.

Reply:
In which case, using `Base.metadata.tables.keys()` might make sense.

Reply:
Okay, switched to table keys. I wanted to try and parameterise this to behave like the `table.delete()` function but it would add too much sqlalchemy internals to be worth it, imo. I don't really like the idea of executing a plain string statement but the benefits of clarity seem to outweigh that consideration in this case.

Reply:
I think this the right way. In postgress we need to first delete all the data with foreign keys by executing `table.delete()` on `reversed(Base.metadata.sorted_tables)` before we can execute `Base.metadata.drop_all()`.

Reply:
@sayanarijit I don't think you'd need to delete the entries first in order to drop the tables, would we? Certainly, `drop_all()` worked in the tests, but they might not contain all the possible data structures. I could certainly see there being problems dropping only a subset of tables without CASCADE.

Either way, I've made the switch to TRUNCATE.

Reply:
Probalby right... I was having this issue by running drop_all() directly while migrating from sqlite to pgsql. It might be caused by cascade not properly set in some tables (intentionally or unintentionally).